cpp/include/cudf/detail/utilities/vector_factories.hpp

```diff
@@ -281,6 +281,20 @@ rmm::device_uvector<typename Container::value_type> make_device_uvector_sync(
   return make_device_uvector_sync(device_span<typename Container::value_type const>{c}, stream, mr);
 }
 
+namespace {
+
+// Utility function template to allow copying to either a thrust::host_vector or std::vector
+template <typename T, typename OutContainer>
+OutContainer make_vector_async(device_span<T const> v,
```

Comment:
Good point. I went ahead and made the `stream` parameter to all the `async` factories required rather than optional. This uncovered two places in `concatenate.cu` where a stream was not being passed. Surprisingly, there were no other places!